src/forms/userForm.tsx

```diff
@@ -326,60 +273,102 @@ class UserForm extends React.Component<UserFormProps, UserFormState> {
     return 0;
   }
 
-  private renderDatacenterChecks() {
+  private handleAssetPermissionSelection() {
+    var updatedPermissions = this.state.permissions;
+    if (
+      this.state.asset_permission_selection === AssetPermissionSelection.GLOBAL
+    ) {
+      updatedPermissions.site_permissions = this.state.initialValues.site_permissions;
+      updatedPermissions.asset_management = false;
+      this.setState({
+        asset_permission_selection: AssetPermissionSelection.PER_SITE,
+        permissions: updatedPermissions,
+      });
+    } else {
+      let permissions: Array<string>;
+      permissions = [];
+      for (var i = 0; i < this.state.sites.length; i++) {
+        permissions.push(this.state.sites[i].id);
+      }
+      updatedPermissions.site_permissions = permissions;
+      updatedPermissions.asset_management = this.state.show_asset_options;
+      this.setState({
+        asset_permission_selection: AssetPermissionSelection.GLOBAL,
+        permissions: updatedPermissions,
+      });
+    }
+  }
+
+  private renderAssetPermissionDetails() {
+    return (
+      <Card className="permission-card" elevation={Elevation.ZERO}>
+        <RadioGroup
+          inline={true}
+          label="Select asset permission level"
+          onChange={() => {
+            this.handleAssetPermissionSelection();
+          }}
+          selectedValue={this.state.asset_permission_selection}
+        >
+          <Radio
+            label={AssetPermissionSelection.GLOBAL}
+            value={AssetPermissionSelection.GLOBAL}
+          />
+          <Radio
+            label={AssetPermissionSelection.PER_SITE}
+            value={AssetPermissionSelection.PER_SITE}
+          />
+        </RadioGroup>
+        {this.state.asset_permission_selection ===
+        AssetPermissionSelection.PER_SITE
+          ? this.renderSiteChecks()
+          : null}
+      </Card>
+    );
```

Comment:
I like this refactor a lot, way cleaner than what I had initially

Reply:
cheers!